### PR TITLE
feat: payment ledger revamp

### PR DESCRIPTION
**Pull request by SankhaChak**
## :pushpin: References

- **Wonder issue:** https://app.wonderverse.xyz/dashboard?task=69481498003112449&view=grid
- **Related pull-requests:** https://github.com/wondrous-dev/wondrous-backend/pull/612

## :blue_book: Description
Revamp UI ledger page under settings

## :movie_camera: Screenshot or Video
https://www.loom.com/share/ac8c5ef2f9cb4464b4252e2a733410c9

## :cake: Checklist:

- [x] I self-reviewed my changes
- [x] My changes follow the style guide: https://creative-earth-33e.notion.site/TT-Wonder-Style-guide-4702a45133374148953bfcaf584120b7
- [x] I tested my changes in Chrome

### CODE REVIEW

File: wondrous-app/components/Settings/Payouts/hooks.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { useEffect } from 'react';
+import { useLazyQuery } from '@apollo/client';
+import { GET_PAYMENT_METHODS_FOR_ORG } from 'graphql/queries';
+
+export const useGetPaymentMethodsForOrg = (orgId, includeDeactivated = false) => {
```
Comment (terryli0095) [inline wondrous-app/components/Settings/Payouts/hooks.tsx:5]: @vefimov  do we need to use hook here? is there a benefit? genuinely curious

File: wondrous-app/components/Settings/Payouts/constants.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export const INITIAL_SELECTION_OPTIONS = [{ label: 'Select all payments', value: 'all' }];
```
Comment (terryli0095) [inline wondrous-app/components/Settings/Payouts/constants.tsx:1]: @vefimov  is this kind of initial value the right pattern?

**Comment by terryli0095** [comment]
when you do Number.POSITIVE_INFINITY and feed it to graphql, it's just gonna be null.. so weirdly it accidently works. but we should do proper pagination here. probably default limit 30 
Screen Shot 2022-10-14 at 8 59 03 PM

File: wondrous-app/components/Settings/Payouts/styles.tsx [hunk 3]
```diff
@@ -1,11 +1,268 @@
...
+export const StyledTable = styled(Table)`
+  && {
+    width: 100%;
+    margin: 14px 0;
+    border-collapse: collapse;
+  }
+`;
+
+export const StyledTableHead = styled(TableHead)`
+  & .MuiTableCell-head {
```
Comment (vefimov) [inline wondrous-app/components/Settings/Payouts/styles.tsx:180]: `& ` here do nothing

File: wondrous-app/components/Settings/Payouts/styles.tsx [hunk 4]
```diff
@@ -1,11 +1,268 @@
...
+    font-family: 'Space Grotesk';
+    font-size: 14px;
+    font-style: normal;
+    font-weight: 700;
+    letter-spacing: 0em;
+    border-radius: 0px;
+    padding: 10px !important;
+  }
+
+  & .MuiTableCell-head:first-of-type {
```
Comment (vefimov) [inline wondrous-app/components/Settings/Payouts/styles.tsx:193]: `& ` here do nothing

File: wondrous-app/components/Settings/Payouts/styles.tsx [hunk 5]
```diff
@@ -1,11 +1,268 @@
...
+    border-top-left-radius: 6px;
+    border-bottom-left-radius: 6px;
+  }
+
+  & .MuiTableCell-head:last-of-type {
```
Comment (vefimov) [inline wondrous-app/components/Settings/Payouts/styles.tsx:198]: `& ` here do nothing

File: wondrous-app/components/Settings/Payouts/styles.tsx [hunk 6]
```diff
@@ -1,11 +1,268 @@
...
+    border-top-right-radius: 6px;
+    border-bottom-right-radius: 6px;
+  }
+`;
+
+export const StyledTableBody = styled(TableBody)`
```
Comment (vefimov) [inline wondrous-app/components/Settings/Payouts/styles.tsx:204]: you don't need styled component, just write `<TableBody border="none" mt="15px">`

File: wondrous-app/components/Settings/Payouts/styles.tsx [hunk 7]
```diff
@@ -24,23 +281,243 @@ export const TableCellText = styled(Typography)`
...
+    color: ${(props) => (props.disabled ? palette.grey800 : palette.highlightPurple)} !important;
+    opacity: ${(props) => (props.disabled ? 0.5 : 1)};
+  }
+`;
+
+export const PayeeProfileLink = styled.a`
+  text-decoration: none;
+`;
+
+export const PayeePayButton = styled(StyledButton)`
```
Comment (vefimov) [inline wondrous-app/components/Settings/Payouts/styles.tsx:296]: use Button from `components/Button`, if button has new style, add styles to the theme there and update storybook

File: wondrous-app/components/Settings/Payouts/styles.tsx [hunk 8]
```diff
@@ -24,23 +281,243 @@ export const TableCellText = styled(Typography)`
...
+      cursor: ${(props) => (props.disabled ? 'not-allowed' : 'pointer')};
+
+      &:hover {
+        background: ${(props) => (props.disabled ? palette.background.default : 'transparent')};
+      }
+    }
+  }
+`;
+
+export const PayeeUsername = styled(Typography)`
```
Comment (vefimov) [inline wondrous-app/components/Settings/Payouts/styles.tsx:324]: use `sx` prop

File: wondrous-app/components/Settings/Payouts/styles.tsx [hunk 9]
```diff
@@ -24,23 +281,243 @@ export const TableCellText = styled(Typography)`
...
+  && {
+    font-family: ${typography.fontFamily};
+    font-size: 13px;
+    font-weight: 700;
+    color: ${palette.white};
+  }
+`;
+
+export const PayeeAddressTagContainer = styled.div`
```
Comment (vefimov) [inline wondrous-app/components/Settings/Payouts/styles.tsx:333]: use `Grid` from mui

File: wondrous-app/components/Settings/Payouts/styles.tsx [hunk 10]
```diff
@@ -24,23 +281,243 @@ export const TableCellText = styled(Typography)`
...
+
+export const BottomActionBarText = styled(Typography)`
+  && {
+    font-family: ${typography.fontFamily};
+    font-size: 13px;
+    color: ${palette.grey250};
+  }
+`;
+
+export const BottomActionBarPayButton = styled(PayeePayButton)`
```
Comment (vefimov) [inline wondrous-app/components/Settings/Payouts/styles.tsx:449]: I'm sure that you don't need custom styles and can use `components/Button`

File: wondrous-app/components/Settings/Payouts/styles.tsx [hunk 11]
```diff
@@ -24,23 +281,243 @@ export const TableCellText = styled(Typography)`
...
+      transition: background 0.2s ease-in-out;
+
+      &:hover {
+        background: transparent;
+      }
+    }
+  }
+`;
+
+export const BottomActionBarButton = styled(Button)`
```
Comment (vefimov) [inline wondrous-app/components/Settings/Payouts/styles.tsx:465]: I'm sure that you don't need custom styles and can use `components/Button`

File: wondrous-app/components/Settings/Payouts/styles.tsx [hunk 12]
```diff
@@ -24,23 +281,243 @@ export const TableCellText = styled(Typography)`
...
+  background: ${palette.grey85};
+  border-radius: 6px;
+
+  svg {
+    width: 20px;
+    height: 20px;
+  }
+`;
+
+export const BottomActionBarMultipleChainSelectedErrorText = styled(Typography)`
```
Comment (vefimov) [inline wondrous-app/components/Settings/Payouts/styles.tsx:495]: use `sx` prop

File: wondrous-app/components/Settings/Payouts/styles.tsx [hunk 13]
```diff
@@ -24,23 +281,243 @@ export const TableCellText = styled(Typography)`
...
+      font-weight: 700;
     }
-    input[type='checkbox'][disabled] {
-      background: red;
-      -webkit-appearance: none;
-      -moz-appearance: none;
-      height: 12px;
-      width: 12px;
+
+    button {
```
Comment (vefimov) [inline wondrous-app/components/Settings/Payouts/styles.tsx:506]: you can use `components/Button`, run `yarn storybook` to see UI and all props

**Review by vefimov — CHANGES_REQUESTED** [review]
- please check storybook
- read this article https://mui.com/system/properties/
- read this article https://mui.com/material-ui/customization/how-to-customize/

File: wondrous-app/components/Settings/Payouts/styles.tsx [hunk 14]
```diff
@@ -1,11 +1,268 @@
...
+    border-color: ${(props) => (props.isSelected ? palette.highlightPurple : palette.black98)};
+    border-radius: 4px;
+    font-weight: ${(props) => (props.isSelected ? 500 : 400)};
+    display: flex;
+    align-items: center;
+    gap: 10px;
+    padding: 10px;
+    transition: background 0.2s ease-out;
+
+    :hover {
```
[the remark this answers is not in the document]
Comment (SankhaChak) [reply wondrous-app/components/Settings/Payouts/styles.tsx:132]: both works

File: wondrous-app/components/Settings/Payouts/styles.tsx (hunk 9 above)
Comment (SankhaChak) [reply wondrous-app/components/Settings/Payouts/styles.tsx:333]: I believe Grid doesn't support `cursor: pointer`

File: wondrous-app/components/Settings/Payouts/styles.tsx (hunk 6 above)
Comment (SankhaChak) [reply wondrous-app/components/Settings/Payouts/styles.tsx:204]: doesn't support border / margin props

File: wondrous-app/components/Settings/Payouts/styles.tsx (hunk 12 above)
Comment (SankhaChak) [reply wondrous-app/components/Settings/Payouts/styles.tsx:495]: for this use case, it feels simplified to use a custom styled component

File: wondrous-app/components/Button/index.tsx [hunk 15]
```diff
@@ -32,7 +32,10 @@ const ButtonInner = styled.button`
   font-weight: ${(props) => props.fontWeight || 400};
   font-family: ${typography.fontFamily};
   justify-content: center;
-  padding: 0 ${(props) => props.paddingX || 30}px;
```
Comment (terryli0095) [inline wondrous-app/components/Button/index.tsx:35]: won't this break other part code?

File: wondrous-app/components/Button/index.tsx [hunk 16]
```diff
@@ -32,7 +32,10 @@ const ButtonInner = styled.button`
...
+  padding-left: ${(props) => props.paddingX || 30}px;
+  padding-right: ${(props) => props.paddingX || 30}px;
+  padding-top: ${(props) => props.paddingY || 0}px;
+  padding-bottom: ${(props) => props.paddingY || 0}px;
```
Comment (SankhaChak) [inline wondrous-app/components/Button/index.tsx:38]: compensating `padding: 0 ${(props) => props.paddingX || 30}px;` @terryli0095

**Pull request merged**